Objects/typeobject.c

```diff
@@ -5971,13 +5973,22 @@ hackcheck(PyObject *self, setattrofunc func, const char *what)
         } else if (base->tp_setattro != slot_tp_setattro) {
             /* 'base' is not a Python class and overrides 'func'.
                Its tp_setattro should be called instead. */
-            PyErr_Format(PyExc_TypeError,
-                         "can't apply this %s to %s object",
-                         what,
-                         type->tp_name);
-            return 0;
+            should_break = 1;
+        } else {
+            /* we found another class in the MRO that overrides the tp_setattro */
+            should_break = 0;
         }
+
     }
+    if (should_break == 1) {
```

Comment:
This is correct only when `should_break = 1;` would be set before `should_break = 0;`. If you change base classes order, `1` could be set after setting `0`.

Reply:
yep, my bad, fixing

Reply:
would a `should_break = 0 + should_break` be enough? or there's any better way of doing it? (I'm quite rusty in C)

Reply:
Actually, just a break after should be enough (if we found at least one class that overrides the `__setattr__` we are good to go no matter what).

Reply:
Unfortunately, I'm not sure what exactly this code is expected to check (I reported the bug, so I saw your PR).

If you do  `should_break = 0 + should_break`, it will be the same as previously - if `should_break` get `1` any time, it will be left here. I think the easiest way to do what you try is to have two variables and check after for (one to mark possible break, second to mark 'we found another class in the MRO that overrides the tp_setattro'), but it doesn't look pretty to me.

Reply:
Just updated the PR, so instead of iterating over the whole MRO and keeping track if we should break or not, I'm breaking not the first time that we find one class that has a custom `tp_setattro`, as if there's any class in the MRO that overrides it is good enough :)